sdk/internal/rpcerrors/errors.go

```diff
@@ -9,33 +9,82 @@ import (
 	"go.autokitteh.dev/autokitteh/sdk/sdkerrors"
 )
 
-func TranslateError(err error) error {
-	var err1 error
-	switch connect.CodeOf(err) {
+func ToSDKError(err error) error {
+	if err == nil {
+		return err
+	}
+
+	var sdkErr error
+	var connectErr *connect.Error
+
+	if !errors.As(err, &connectErr) { // not a connect error?
+		return err
+	}
+
+	// convert connect errors to sdk. Their strings are almost identical
+	switch connectErr.Code() {
 	case connect.CodeAlreadyExists:
-		err1 = sdkerrors.ErrAlreadyExists
+		sdkErr = sdkerrors.ErrAlreadyExists
 	case connect.CodeNotFound:
-		err1 = sdkerrors.ErrNotFound
+		sdkErr = sdkerrors.ErrNotFound
 	case connect.CodeInvalidArgument:
-		err1 = sdkerrors.ErrInvalidArgument{Underlying: err}
+		sdkErr = sdkerrors.ErrInvalidArgument{Underlying: err}
 	case connect.CodeUnimplemented:
-		err1 = sdkerrors.ErrNotImplemented
+		sdkErr = sdkerrors.ErrNotImplemented
 	case connect.CodeUnauthenticated:
-		err1 = sdkerrors.ErrUnauthenticated
+		sdkErr = sdkerrors.ErrUnauthenticated
 	case connect.CodePermissionDenied:
-		err1 = sdkerrors.ErrUnauthorized
+		sdkErr = sdkerrors.ErrUnauthorized
 	case connect.CodeResourceExhausted:
-		err1 = sdkerrors.ErrLimitExceeded
+		sdkErr = sdkerrors.ErrLimitExceeded
+	case connect.CodeUnknown: // returned as connect.Error, but unrelated to RPC, just unwrap underlying error
+		return connectErr.Unwrap()
 	default:
-		err1 = sdkerrors.ErrRPC
+		sdkErr = sdkerrors.ErrRPC
 	}
 
-	if cerr := new(connect.Error); errors.As(err, &cerr) {
-		return fmt.Errorf("%w: %s (%v)", err1, cerr.Message(), cerr.Details())
+	// err is a connect error (checked in connect.CodeOf), so we can safely cast it
+	connErr := err.(*connect.Error)
```

Comment:
I think you removed the connect.CodeOf check so is this still valid conversion ?
also, why do we need this part if you already converted it before to sdk error ?

Reply:
yes, conversion is done above in errors.As. There err is converted to connect.Error. If failed we return an original error

Reply:
ah. I see. you are right. unneeded conversion